packages/compass-aggregations/src/components/stage-editor-toolbar/stage-operator-select.jsx

```diff
@@ -1,61 +1,82 @@
-import React, { PureComponent } from 'react';
+import React, { useCallback, useMemo } from 'react';
 import PropTypes from 'prop-types';
-import Select from 'react-select-plus';
 import { connect } from 'react-redux';
 import { changeStageOperator } from '../../modules/pipeline-builder/stage-editor';
-import SelectOptionWithTooltip from './select-option-with-tooltip/select-option-with-tooltip';
 import { filterStageOperators } from '../../utils/stage';
 
-import styles from './stage-operator-select.module.less';
+import { Combobox, ComboboxOption, css, spacing } from '@mongodb-js/compass-components';
+import { isAtlasOnly } from '../../utils/stage';
 
-/**
- * Select from a list of stage operators.
- */
+const inputWidth = spacing[7] * 2;
+const descriptionWidth = spacing[5] * 14;
+const stageNameWidth = spacing[4] * 8;
+const dropdownWidth = stageNameWidth + descriptionWidth;
 
-export class StageOperatorSelect extends PureComponent {
-  static propTypes = {
-    index: PropTypes.number.isRequired,
-    stages: PropTypes.array.isRequired,
-    selectedStage: PropTypes.string,
-    onChange: PropTypes.func.isRequired,
-    isDisabled: PropTypes.bool,
+const comboboxStyles = css({
+  marginLeft: spacing[2],
+  width: inputWidth,
+  '& [role="combobox"]': {
+    height: spacing[4] - 2 // match other xs controls
   }
+});
 
-  /**
-   * Called when the stage operator is selected.
-   * @param {String} name The name of the stage operator.
-   * @returns {void}
-   */
-  onStageOperatorSelected = (name) => {
-    this.props.onChange(this.props.index, name);
-  }
-
-  /**
-   * Render the stage operator select component.
-   *
-   * @returns {Component} The component.
-   */
-  render() {
-    return (
-      <div className={styles['stage-operator-select']}>
-        <Select
-          optionComponent={SelectOptionWithTooltip}
-          simpleValue
-          searchable
-          openOnClick
-          openOnFocus
-          clearable={false}
-          disabled={this.props.isDisabled}
-          className={styles['stage-operator-select-control']}
-          options={this.props.stages}
-          value={this.props.selectedStage}
-          onChange={this.onStageOperatorSelected}
-        />
-      </div>
-    );
-  }
+function comboboxOptionStyles(stage) {
+  return css({
+    '&::after': {
+      content: JSON.stringify(
```

Comment:
Just a thought, it might be worth leaving a comment why we show the combobox description inside of css. Nice workaround btw!